### PR TITLE
[DOCS-2848] [CWS] documentation around troubleshooting, starting with self-tests

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR creates a page (that can be incorporated elsewhere) describing how to use the self-test feature of the security agent.

### Open questions

- do we want this documentation:
  - appended in another file
  - in a separate file (like in this PR) tilted `Self Tests`, this will need changes to add this page to the nav/sidebar and to the `read_more` links
  - in a separate file tilted `Troubleshooting` that will group all documentation helpful for users trying to debug, ensure that CWS is working correctly in their environments

### Motivation

Documentation around a feature.

### Preview

https://docs-staging.datadoghq.com/paulcacheux/doc-cws-self-test
https://docs-staging.datadoghq.com/paulcacheux/doc-cws-self-test/security_platform/cloud_workload_security/troubleshooting/

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+| Host         | `sudo /opt/datadog-agent/embedded/bin/security-agent runtime self-test`             |
+
+The self-test procedure creates some temporary files and rules monitoring
+them, and then triggers those rules ensuring that events are correctly propagated.
+
+If everything is working as expected, you should see:
+```
+Runtime self test: OK
+```
+
```
[the remark this answers is not in the document]
Comment (paulcacheux) [reply content/en/security_platform/cloud_workload_security/self_test.md:29]: Good point I will add this

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+kind: documentation
+description: "Troubleshooting guide for Datadog Cloud Workload Security."
+---
+
+## Self tests
+
+In order to ensure that:
+- the communication between the `security-agent` and the `system-probe` is
+working as expected
+- the Cloud Workload Security probe is able to respond to system events
```
Comment (safchain) [inline content/en/security_platform/cloud_workload_security/self_test.md:12]: s/respond/detect/ WDYT ?

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+title: Troubleshooting
+kind: documentation
+description: "Troubleshooting guide for Datadog Cloud Workload Security."
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:4]:
```suggestion
description: "Troubleshooting for Cloud Workload Security."
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+## Self tests
+
+In order to ensure that:
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:9]:
```suggestion
In order to ensure that the communication between the `security-agent` and the `system-probe` is working as expected and the Cloud Workload Security probe is able to detect system events, you can manually trigger self tests by running the following command:

```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+- the communication between the `security-agent` and the `system-probe` is
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:10]:
```suggestion
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+working as expected
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:11]:
```suggestion
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+- the Cloud Workload Security probe is able to detect system events
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:12]:
```suggestion
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+you can manually trigger self tests by running the following command:
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:14]:
```suggestion
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+| Platform     | Command                                                                             |
+| --------     | -------                                                                             |
+| Docker       | `docker exec -it datadog-agent security-agent runtime self-test`                    |
+| Kubernetes   | `kubectl exec -it <POD_NAME> -c security-agent -- security-agent runtime self-test` |
+| Host         | `sudo /opt/datadog-agent/embedded/bin/security-agent runtime self-test`             |
+
+The self-test procedure creates some temporary files and rules monitoring
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:22]:
```suggestion
The self-test procedure creates some temporary files and rules to monitor them, and then triggers those rules to ensure that events are correctly propagated.
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+them, and then triggers those rules ensuring that events are correctly propagated.
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:23]:
```suggestion
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+If everything is working as expected, you should see:
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:25]:
```suggestion
The following response appears when rules are propagated.
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+```
+Runtime self test: OK
+```
+and then see events coming from the `runtime-security-agent` in your logs
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:29]:
```suggestion

You can now see events coming from the `runtime-security-agent` in the Log Explorer.
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+{{< img src="security_platform/cws/self_test_logs.png" alt="Self Test events in Logs explorer" style="width:90%;">}}
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:31]:
```suggestion
{{< img src="security_platform/cws/self_test_logs.png" alt="Self test events in the Log Explorer" style="width:90%;">}}
```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+In the case of an error, you should see:
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:33]:
```suggestion
If not properly configured, the following error may occur, where `<ERROR_MESSAGE>` is a description of the error encountered during the test.

```

File: content/en/security_platform/cloud_workload_security/self_test.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+```
+Runtime self test: <ERROR_MESSAGE>
+```
+where `<ERROR_MESSAGE>` is a description of the error encountered during the test.
```
Comment (sarina-dd) [inline content/en/security_platform/cloud_workload_security/self_test.md:37]:
```suggestion
```

File: content/en/security_platform/cloud_workload_security/self_test.md (hunk 14 above)
Comment (sarina-dd) [reply content/en/security_platform/cloud_workload_security/self_test.md:33]: Do we have common errors / resolution steps for those errors? Might be helpful to add in a few here with mention of how to resolve. If not, I'd remove this section entirely as it's self-explanatory and could spark more confusion than help.

**Review by sarina-dd — COMMENTED** [review]
Editorial review completed

File: content/en/security_platform/cloud_workload_security/self_test.md (hunk 14 above)
Comment (paulcacheux) [reply content/en/security_platform/cloud_workload_security/self_test.md:30]: For now we don't. I will remove this section

**Review by safchain — APPROVED** [review]

**Review by n06 — APPROVED** [review]
LGTM

**Pull request merged**